tidy3d/components/geometry.py

```diff
@@ -1090,6 +1165,46 @@ def bounds(self) -> Bound:
         coord_max = tuple(c + self.radius for c in self.center)
         return (coord_min, coord_max)
 
+    def volume(self, bounds: Bound) -> float:
+        """Returns object's volume. If the object extends beyound bounds then
+        the computed volume may be only a rough approximation
+
+        Returns
+        -------
+        float
+            Volume.
+        """
+        volume = 4.0 / 3.0 * np.pi * self.radius**3
+
+        # a very loose upper bound on how much of sphere is in bounds
```

Comment:
There's a formula for this, maybe we could use it here in some more simple cases? https://en.wikipedia.org/wiki/Spherical_cap